src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1719,7 +1723,10 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				this.xterm.raw.options.disableStdin = false;
 				this._isExiting = false;
 			}
-			this.xterm.clearDecorations();
+			//TODO:@meganrogge does this break anything?
+			if (this.statusList.statuses.find(s => s.id === TerminalStatus.RelaunchNeeded)) {
+				this.xterm.clearDecorations();
+			}
```

Comment:
Does this work for seamless restart and tasks?

```suggestion
			//TODO:@meganrogge does this break anything?
			if (reset) {
				this.xterm.clearDecorations();
			}
```